File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.py
'''
F(n) = F(n-1) + F(n-2)

a[0] = 0, a[1] = 1 초기화 시켜줌

for문을 돌려 n번까지 더해줌
'''
def solution(n):
    answer = 0
    a = [0] * (n+1)
    a[0] = 0
    a[1] = 1
    # print(a)
    for i in range(2,n+1 ) :
        a[i] += a[i-1] + a[i-2]
        answer = a[n] %1234567
       
    
    return answer

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.java
class Solution {
    public int solution(int n) {
        // if (n == 0) return 0;
        // if (n == 1) return 1;
        
        long[] n_list = new long[n + 1];
        n_list[0] = 0;
        n_list[1] = 1;

        for (int i = 2; i <= n; i++) {
            n_list[i] = (n_list[i - 1] + n_list[i - 2]) % 1234567;
        }
        
        return (int) n_list[n];
    }
}
